src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -67,7 +67,13 @@ class BlockExecution(
     * @param block the block with transactions to run
     */
   private[ledger] def executeBlockTransactions(block: Block): Either[BlockExecutionError, BlockResult] = {
-    val parentStateRoot = blockchain.getBlockHeaderByHash(block.header.parentHash).map(_.stateRoot)
+    val parentStateRoot =
+      blockchain
+        .getBlockHeaderByHash(block.header.parentHash)
+        .map(_.stateRoot)
+        .getOrElse(
+          throw new IllegalStateException("Cannot find parent")
```

Comment:
Maybe it would be better to return `Left(BlockExecutionError)` even in that case?